src/foam/nanos/notification/email/DAOResourceLoader.java

```diff
@@ -125,21 +132,39 @@ public static EmailTemplate findTemplateHelper(X x, String name, String groupId,
 
     if ( emailTemplate_ != null ) {
       EmailTemplate clonedTemplate = (EmailTemplate) emailTemplate_.fclone();
-      String sourceType = (String) templateArgs.get("templateSourceType");
-      String templateSource = (String) templateArgs.get("templateSource");
-      String source = templateSource != null ? templateSource : "emailTemplate";
+      setTemplateSource(x, name, clonedTemplate,templateArgs);
 
-      clonedTemplate.setSourceClass(source);
+      emailTemplateDAO.put(clonedTemplate);
+    }
 
-      PM pm = PM.create(x, source,  "emailTemplate: " + name);
-      pm.log(x);
+    return emailTemplate_;
+  }
+
+  public static EmailTemplate findTemplateByName(X x, String name, Map templateArgs) {
+    DAO emailTemplateDAO = (DAO) x.get("localEmailTemplateDAO");
+    EmailTemplate emailTemplate_ = (EmailTemplate) emailTemplateDAO
+      .find(EQ(EmailTemplate.NAME, name));
+
+    if ( emailTemplate_ != null ) {
+      EmailTemplate clonedTemplate = (EmailTemplate) emailTemplate_.fclone();
+      setTemplateSource(x, name, clonedTemplate,templateArgs);
 
       emailTemplateDAO.put(clonedTemplate);
     }
 
     return emailTemplate_;
   }
 
+  public static void setTemplateSource(X x, String name, EmailTemplate emailTemplate, Map templateArgs) {
+      String sourceType = (String) templateArgs.get("templateSourceType");
+      String templateSource = (String) templateArgs.get("templateSource");
+      String source = templateSource != null ? templateSource : "emailTemplate";
+      emailTemplate.setSourceClass(source);
+
+      PM pm = PM.create(x, source,  "emailTemplate: " + name);
```

Comment:
```suggestion
      PM pm = PM.create(x, source,  "emailTemplate", name);
```
PM will concatenate arguments with : for you.